### PR TITLE
fix(incsearch): viewport is not updating during searching

### PR DESCRIPTION
**Pull request by xiyaowong**
- Resolve: #1149
- Resolve: #1555

### CODE REVIEW

File: src/viewport_manager.ts [hunk 1]
```diff
@@ -87,6 +93,27 @@ export class ViewportManager implements Disposable {
                         view.line = curline;
                         view.col = curcol;
                     }
```
Comment (justinmk) [inline src/viewport_manager.ts:95]: off-topic, line 83 (89) looks weird, is it intentional?

                        for (const [grid, , topline, botline, curline, curcol] of args) {

Comment (xiyaowong) [reply src/viewport_manager.ts:95]: Yes, currently the second item(win) is not needed

**Comment by theol0403** [comment]
This feels like a hack, and I don't really like the added code. Is there any other possible fix that is more generic?

**Comment by xiyaowong** [comment]
Neovim did not send the corresponding grid information during incsearch, and no relevant instructions were found in the neovim documentation. @justinmk  

This 'hack' is currently the simplest one among the solutions I have tried.

**Comment by theol0403** [comment]
It used to work before your highlight manager refactor... do you know why it broke?

**Comment by xiyaowong** [comment]
No, this is a problem that has always existed. I tested 0.3.2 💆🏻‍♂️

It is fundamentally unrelated to highlighting.

https://github.com/vscode-neovim/vscode-neovim/issues/1149#issuecomment-1427386558

**Review by theol0403 — APPROVED** [review]

**Comment by xiyaowong** [comment]
This problem can only be easily reproduced in clean mode. Most people(including me) don't encounter this problem because they use a search plugin or execute nohl after searching

**Comment by xiyaowong** [comment]
https://github.com/vscode-neovim/vscode-neovim/pull/1453

**Comment by xiyaowong** [comment]
https://github.com/neovim/neovim/pull/27858

**Pull request merged**